dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -77,8 +75,18 @@ internal class RumSessionScope(
 
             onApplicationDisplayed(event, viewScope, actualWriter)
             activeChildrenScopes.add(viewScope)
-        } else if (activeChildrenCount == 0) {
-            devLogger.w(MESSAGE_MISSING_VIEW)
+        } else if (activeChildrenScopes.size == 0 && (
+                event is RumRawEvent.AddError ||
+                    event is RumRawEvent.AddLongTask ||
+                    event is RumRawEvent.StartAction ||
+                    event is RumRawEvent.StartResource)
+        ) {
+            // there is no active ViewScope to handle this event. We will assume the application
+            // is in background and we will create a RumBackgroundScope to handle all the events.
```

Comment:
I'm not sure on this assumption. It is very liberal and will also trigger in those scenarios:
* incorrect manual instrumentation - if a user forgets to start a view, but sends something with `Global.rum`,
* missing auto-instrumentation - if a user forgets to enable views auto-instrumentation but enables for example actions instrumentation (quite a common issue among iOS SDK users who report "no RUM data is sent"),
* if a resource is started shortly after `Datadog.initialize()` but before starting first auto-instrumented view.

In general - all cases where the app is still in foreground, but has no view.

Can't we make background views only spanned on explicit "app leaves foreground" / "app enters foreground" callbacks? Or there are no such callbacks on Android?

Reply:
The way we decided in the `RFC` was to start a background view if a resource, action, error event was reported without an active view. The idea is not to loose all these events. Regarding your last remark, I am afraid in Android is rather more complicated than in `iOS` to detect this, we have a `ProcessLifecycleMonitor` which kind of detects that but it will not be very precise. I think those corner cases can be accepted in this case and will allow this feature with less complexity, what do you think ?

Reply:
I see 👌. Indeed those are edge cases and maybe showing these events in the RUM Explorer with wrong (background view) assignment is better than loosing them at all. We could do similar on iOS and depend on view presence (instead of foreground activity notifications) for having similar experience 👍.

Reply:
so basically `background` scope is becoming kind of `default uncaught action handler` if we build analogy with `default uncaught exception handler`? Indeed there are some cases when it may catch events when app is not in the background, but I agree that the requirement of minimising the rate of missed events is more important.

How do we handle the case when app goes into foreground after being in background? What will stop background scope?

Reply:
The new view event from foreground, it is automatically handled.